portality/tasks/harvester_helpers/epmc/client.py

```diff
@@ -102,45 +103,54 @@ def complex_search(cls, query_builder, cursor="", page_size=25):
         return cls.query(query_builder.to_url_query_param(), cursor=cursor, page_size=page_size)
 
     @classmethod
-    def complex_search_iterator(cls, query_builder, page_size=1000, throttle=None):
-        return cls.iterate(query_builder.to_url_query_param(), page_size=page_size, throttle=throttle)
+    def complex_search_iterator(cls, query_builder, job, page_size=1000, throttle=None):
+        return cls.iterate(query_builder.to_url_query_param(), job, page_size=page_size, throttle=throttle)
 
     @classmethod
-    def iterate(cls, query_string, page_size=1000, throttle=None):
+    def iterate(cls, query_string, job, page_size=1000, throttle=None):
+        cls.job = job
         cursor = ""
         last = None
         while True:
             if last is not None and throttle is not None:
                 diff = (datetime.utcnow() - last).total_seconds()
-                app.logger.debug("Last request at {x}, {y}s ago; throttle {z}s".format(x=last, y=diff, z=throttle))
+                cls.job.add_audit_message("Last request at {x}, {y}s ago; throttle {z}s".format(x=last, y=diff, z=throttle))
                 if diff < throttle:
                     waitfor = throttle - diff
-                    app.logger.debug("Throttling EPMC requests for {x}s".format(x=waitfor))
+                    cls.job.add_audit_message("Throttling EPMC requests for {x}s".format(x=waitfor))
                     time.sleep(waitfor)
-            results, cursor = cls.query(query_string, cursor=cursor, page_size=page_size)
+            results, cursor = cls.query(query_string, job, cursor=cursor, page_size=page_size)
             last = datetime.utcnow()
             if len(results) == 0:
                 break
             for r in results:
                 yield r
 
     @classmethod
-    def query(cls, query_string, cursor="", page_size=25):
-        """
-        :return: (results, next_cursor)
-        """
+    def url_from_query(cls, query_string, cursor, page_size):
```

Comment:
This diff is very weirdly presented. All that happened here is the extraction of url_from_query method outside of query method.